src/Material/Textfield.elm

```diff
@@ -349,8 +366,20 @@ view lift model options =
         Nothing -> []
 
 
+    -- These events require special handling because they do not bubble
+    -- Meaning our event handlers for these events have to be on the input-element as well
+    specialEvents =
+      ["blur", "focus"]
+
+    (special, rest) =
+      Dict.partition
```

Comment:
I think it'd be nicer if you split out focus and blur already in config, then let the `on` property either put handler in general listeners or special places for blur/focus. This avoids the slightly heavy `Dict.partition`. 

(You could also argue that this is totally irrelevant and leave it as is.)
(I'm starting to be a bit concerned about overhead of `Options` in general; maybe I'm overreacting.)